Sources/Crust/Parser.swift

```diff
@@ -407,7 +407,21 @@ extension Parser {
     let recordTok = try consume(.recordKeyword)
     let recName = try parseIdentifierToken()
     let paramList = try parseTypedParameterList()
-    let indices = peek() == .colon ? try parseTypeIndices() : nil
+    let indices: TypeIndicesSyntax
+
+    // If we see a semicolon or 'where' after the identifier, the
+    // user likely forgot to provide indices for this data type.
+    // Recover by inserting `: Type`.
+    if [.semicolon, .whereKeyword].contains(peek()) {
```

Comment:
Please factor this into a helper function

Reply:
Sounds good.

Reply:
Done